File: backend/src/main/java/com/efacility/ticketing/service/AuthService.java
package com.efacility.ticketing.service;

import com.efacility.ticketing.dto.AuthResponse;
import com.efacility.ticketing.dto.request.LoginRequest;
import com.efacility.ticketing.dto.request.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: frontend/src/api/authService.ts
import api from '../config/api';
import type { LoginRequest, RegisterRequest, AuthResponse } from '../types/auth.types';
import type { BackendResponse } from '../types/api.types';

export const authService = {
  async login(data: LoginRequest): Promise<AuthResponse> {
    const res = await api.post<BackendResponse<{ auth: AuthResponse }>>('/auth/login', data);
    return res.data.data.auth;
  },

  async register(data: RegisterRequest): Promise<AuthResponse> {
    const res = await api.post<BackendResponse<{ auth: AuthResponse }>>('/auth/register', data);
    return res.data.data.auth;
  },
};
